### ISSUE TITLE
Azure Auth: failed to decode pkcs12 certificate while creating spt: pkcs12

### ISSUE THREAD

**Issue by onionhammer** [dapr/components-contrib#2396]
## Expected Behavior
Creds file should work

## Actual Behavior
Creds file fails with:

```
process component localsecrets error: [INIT_COMPONENT_FAILURE]: initialization error occurred for send-email (bindings.azure.storagequeues/v1): invalid credentials with error: failed to decode pkcs12 certificate while creating spt: pkcs12: error reading P12 data: asn1: structure error: tags don't match (16 vs {class:0 tag:13 length:45 isCompound:true}) {optional:false explicit:false application:false private:false defaultValue:<nil> tag:<nil> stringType:0 timeType:0 set:false omitEmpty:false} pfxPdu @2" app_id=c2c-exchange-actors instance=COMP1 scope=dapr.runtime type=log ver=1.9.5
```

Seems to be an issue referenced here: https://github.com/Azure/azure-sdk-for-go/issues/5412

## Steps to Reproduce the Problem

Create a credential with certificate, following this guide: https://docs.dapr.io/developing-applications/integrations/azure/authenticating-azure/#creating-an-azure-ad-application
```
az ad sp create-for-rbac --name ... --role contributor ...  --create-cert 
```

Copy the file from `fileWithCertAndPrivateKey` to home directory, reference it in component.yml

```yml
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: send-email
spec:
  type: bindings.azure.storagequeues
  version: v1
  metadata:
  - name: accountName
    secretKeyRef:
      name: storageAccount
  - name: queueName
    value: send-email
  - name: azureTenantId
    secretKeyRef:
      name: azure__tenantId
  - name: azureClientId
    secretKeyRef:
      name: azure__clientId
  - name: azureCertificateFile
    value: "C:\\Users\\myuser\\.dapr\\certs\\myfile.pem"
```

Run application with dapr

## Release Note

RELEASE NOTE:

**Comment by berndverst** [comment]
@onionhammer reading through the code only PFX certificates (as provided by Keyvault directly) are supported. Fortunately you can take the cert file exported for the service principal and make it a PKCS12 certificate.

Here is what you can do. Open the PEM file and save the actual X509 cert (top part) to one file (cert.pem) and the private key (key.pem) or something like that.

Then create the PKCS12 cert without password.

`openssl pkcs12 -export -out cert.pkcs12 -in cert.pem -inkey key.pem -passout pass: -nokeys`

The resulting PKCS12 should work for you as a certificate. (If it doesn't try base64 encoding it)


@ItalyPaleAle here is the code for the Track 1 SDKs (probably similar in Track2) that only deals with PKCS12 certs.

See https://github.com/Azure/go-autorest/blob/main/autorest/azure/auth/auth.go#L643

https://github.com/Azure/go-autorest/blob/9038e4a609b1899f0eb382d03c3e823b70537125/autorest/adal/persist.go#L88-L135

**Comment by berndverst** [comment]
@onionhammer we'll try to support this weird certificate format natively in 1.11, but for now please create the cert files manually per my instructions.

**Comment by ItalyPaleAle** [comment]
Fixed by #2462

**Comment by vlardn** [comment]
> @onionhammer reading through the code only PFX (PKCS12) certificates (as provided by Keyvault directly) are supported. Fortunately you can take the cert file exported for the service principal and make it a PKCS12 certificate.
> 
> Here is what you can do. Open the PEM file and save the actual X509 cert (bottom part) to one file (cert.pem) and the private key (top part) to another file (key.pem) or something like that.
> 
> Then create the PKCS12 cert without password.
> 
> `openssl pkcs12 -export -out cert.pkcs12 -in cert.pem -inkey key.pem -passout pass: -nokeys`
> 
> The resulting PKCS12 should work for you as a certificate. (If it doesn't try base64 encoding it)
> 
> @ItalyPaleAle here is the code for the Track 1 SDKs (probably similar in Track2) that only deals with PKCS12 certs.
> 
> See https://github.com/Azure/go-autorest/blob/main/autorest/azure/auth/auth.go#L643
> 
> https://github.com/Azure/go-autorest/blob/9038e4a609b1899f0eb382d03c3e823b70537125/autorest/adal/persist.go#L88-L135

Please note that `openssl` command must be provided without `-nokeys` option. This option prevents the private key to be exported and Azure auth will not work.
